MemoryMeter.c

```diff
@@ -33,12 +33,17 @@ static void MemoryMeter_updateValues(Meter* this) {
    size_t size = sizeof(this->txtBuffer);
    int written;
 
+   Settings *settings = this->host->settings;
+
    /* shared, compressed and available memory are not supported on all platforms */
    this->values[MEMORY_METER_SHARED] = NAN;
    this->values[MEMORY_METER_COMPRESSED] = NAN;
    this->values[MEMORY_METER_AVAILABLE] = NAN;
    Platform_setMemoryValues(this);
-
+   if (this->mode == GRAPH_METERMODE && !settings->showCachedMemory) {
+      this->values[MEMORY_METER_BUFFERS] = NAN;
+      this->values[MEMORY_METER_CACHE] = NAN;
+   }
```

Comment:
This is not a valid approach.
Switch the display modes quickly and you get:

Backtrace information:
----------------------
 0: 0x7f88eb4d0050  /lib/x86_64-linux-gnu/libc.so.6  (__sigaction+0x40)  [0x7f88eb4d008f]  {signal frame}
 1: 0x7f88eb51ee3c  /lib/x86_64-linux-gnu/libc.so.6  (pthread_key_delete+0x14c)  [0x7f88eb51ee7c]
 2: 0x7f88eb4cffb2  /lib/x86_64-linux-gnu/libc.so.6  (gsignal+0x12)  [0x7f88eb4cffb2]
 3: 0x7f88eb4ba472  /lib/x86_64-linux-gnu/libc.so.6  (abort+0xd3)  [0x7f88eb4ba472]
 4: 0x7f88eb4ba395  /lib/x86_64-linux-gnu/libc.so.6  (__libc_freeres+0)  [0x7f88eb4ba386]
 5: 0x7f88eb4c8eb2  /lib/x86_64-linux-gnu/libc.so.6  (__assert_fail+0x42)  [0x7f88eb4c8eb2]
 6: 0x5600cfde6d1e  ./htop  (Meter_humanUnit+0xee)  [0x5600cfde6d1e]
 7: 0x5600cfde5970  ./htop  (MemoryMeter_display+0xe0)  [0x5600cfde5970]
 8: 0x5600cfde6a16  ./htop  (LEDMeterMode_draw+0x86)  [0x5600cfde6a16]
 9: 0x5600cfde2d08  ./htop  (Header_draw+0x248)  [0x5600cfde2d08]
10: 0x5600cfdf0dc2  ./htop  (ScreenManager_run+0x6e2)  [0x5600cfdf0dc2]
11: 0x5600cfdd93c3  ./htop  (actionSetup+0x43)  [0x5600cfdd93c3]
12: 0x5600cfde51a1  ./htop  (MainPanel_eventHandler+0xf1)  [0x5600cfde51a1]
13: 0x5600cfdf0996  ./htop  (ScreenManager_run+0x2b6)  [0x5600cfdf0996]
14: 0x5600cfddda84  ./htop  (CommandLine_run+0x744)  [0x5600cfddda84]
15: 0x7f88eb4bb24a  /lib/x86_64-linux-gnu/libc.so.6  (__libc_init_first+0x8a)  [0x7f88eb4bb25a]
16: 0x7f88eb4bb305  /lib/x86_64-linux-gnu/libc.so.6  (__libc_start_main+0x85)  [0x7f88eb4bb305]
17: 0x5600cfdd8bf1  ./htop  (_start+0x21)  [0x5600cfdd8bf1]

Reply:
Should not be a problem anymore